src/lfortran/semantics/ast_body_visitor.cpp

```diff
@@ -1629,6 +1771,10 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         body.reserve(al, x.n_body);
         format_statements.clear();
         transform_stmts(body, x.n_body, x.m_body);
+        copy_stmts(x.n_body, x.m_body);
+        handle_active_entry_points();
```

Comment:
Now let's do this, to ensure fast execution in the default case (no entry points):
```suggestion
        if (entry_points_present) {
            copy_stmts(x.n_body, x.m_body);
            handle_active_entry_points();
        }
```
```